### ISSUE TITLE
colexec: add unsupported types

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#42043]
Currently, not all types are supported by the vectorized engine, and this issue will track those (note that we might not want to implement all of them).

- [ ] `INTERVAL`
- [ ] `TIMESTAMPTZ`
- [ ] `COLLATEDSTRING`
- [ ] `ARRAY`
- [ ] `TIME`
- [ ] `JSONB`

**Comment by glennfawcett** [comment]
I have a customer that has a table with ARRAY and JSONB as columns.  These columns are **NOT** used as part of the aggregate query or retrieved, but regardless this query doesn't benefit from vectorized execution.  It would be nice to allow vectorization for this case.

**Comment by awoods187** [comment]
@glennfawcett could you provide an example query? i wonder if we decode the query into an unsupported type. This seems similar to what @jseldess was reporting. You can also try casting the type to a supported type.

**Comment by bladefist** [comment]
We're unable to use vectorization across the board due to missing timestampz.  Can this be prioritized in anyway? It should unlock a lot of performance gains for us.  I don't think we're missing any of the others.

**Comment by awoods187** [comment]
We have merged timestamptz into 20.1 but it is a bit risky to backport it to 19.2. Does everyone one of your tables use it? For all queries?

**Comment by bladefist** [comment]
@awoods187 Yes, essentially our primary table for a user profile uses it so all queries join to that.  We can wait a little bit if 20.1 is coming soon? thank you!
